GUI/QtQuick/ScaleController.cpp

```diff
@@ -19,10 +20,11 @@ namespace {
 ScaleController::ScaleController(QObject* parent)
    : QObject(parent)
 {
-   qreal dpi = QGuiApplication::primaryScreen()->logicalDotsPerInch();
+   const auto screen = QGuiApplication::screens()[QApplication::desktop()->screenNumber(QApplication::activeWindow())];
+   qreal dpi = screen->logicalDotsPerInch();
```

Comment:
I think aspect ratio should be recalculated also when moving the window from one screen to another.